Source/NSStoryboard.m

```diff
@@ -0,0 +1,967 @@
+              NSXMLNode *appNode = [subnodes objectAtIndex: 0];
+              if (appNode != nil)
+                {
+                  NSXMLElement *objects = (NSXMLElement *)[appNode parent];
+                  NSArray *appConsArr = [appNode nodesForXPath: @"connections" error: NULL];
+                  NSXMLNode *appCons = [appConsArr objectAtIndex: 0];
+                  if (appCons != nil)
+                    {
+                      [appCons detach];
+                    }
+                  
+                  NSArray *appChildren = [appNode children];
+                  NSEnumerator *ace = [appChildren objectEnumerator];
+                  NSXMLElement *ae = nil;
+                  
+                  // Move all application children to objects...
+                  while ((ae = [ace nextObject]) != nil)
+                    {
+                      [ae detach];
+                      [objects addChild: ae];
+                    }
+                  
+                  // Remove the appNode
+                  [appNode detach];
+                  
+                  // create a customObject entry for NSApplication reference...
+                  NSXMLNode *appCustomClass = (NSXMLNode *)[(NSXMLElement *)appNode
+                                                               attributeForName: @"customClass"];
+                  customClassString = ([appCustomClass stringValue] == nil) ?
+                    @"NSApplication" : [appCustomClass stringValue];
+                  NSXMLElement *customObject = nil;
+                  
+                  customObject = 
+                    [self _createCustomObjectWithId: @"-3"
+                                          userLabel: @"Application"
+                                        customClass: @"NSObject"];
+                  [child insertChild: customObject
+                             atIndex: 0];
+                  customObject = 
+                    [self _createCustomObjectWithId: @"-1"
+                                          userLabel: @"First Responder"
+                                        customClass: @"FirstResponder"];
+                  [child insertChild: customObject
+                             atIndex: 0];
+                  customObject =
+                    [self _createCustomObjectWithId: @"-2"
+                                          userLabel: @"File's Owner"
+                                        customClass: customClassString];
+                  if (appCons != nil)
+                    {
+                      [customObject addChild: appCons];
+                    }
+                  [child insertChild: customObject
+                             atIndex: 0];
+                  
+
+                  // Add it to the document
+                  [objects detach];
+                  [doc addChild: objects];
+                  
+                  // Assign application scene...
+                  ASSIGN(_applicationSceneId, sceneId);
```

Comment:
Why is this line here and not up there where we get the value?

Reply:
Added it to where we know we are in the application scene, but after declarations.